common/djangoapps/student/tests/tests.py

```diff
@@ -930,3 +932,190 @@ def test_roundtrip_with_unicode_course_id(self):
+                        ]
+                    }
+                ],
+                'subtitle': 'Dummy program 1 for testing',
+                'name': 'First Program'
+            }
+
+        return programs
+
+    @ddt.data(
+        ('active', [{'sku': ''}, {'sku': ''}, {'sku': ''}, {'sku': ''}], 4, 'slug1'),
+        ('active', [{'sku': ''}, {'sku': ''}, {'sku': ''}], 3, 'slug2'),
+        ('active', [], 0, ''),
+        ('inactive', [{'sku': ''}, {'sku': ''}, {'sku': ''}, {'sku': ''}], 4, 'slug3'),
+    )
+    @ddt.unpack
+    def test_get_xseries_programs_method(self, status, run_modes, run_modes_count, slug):
+        with patch('student.views.dummy_method_for_test_cases') as mock_data:
+            mock_data.return_value = {
+                unicode('course-v1:ManTestX+ManTest2+2014'): {
+                    'category': 'xseries',
+                    'status': status,
+                    'course_codes': [
+                        {
+                            'organization': {'display_name': 'Test Organization 1', 'key': 'edX'},
+                            'marketing_slug': slug,
+                            'display_name': 'Manual Test',
+                            'key': 'TEST_A',
+                            'run_modes': run_modes
+                        }
+                    ],
+                    'subtitle': 'Dummy program 1 for testing',
+                    'name': 'First Program'
+                }
+            }
+            parse_data = _get_xseries_programs(
+                self.user, [unicode('course-v1:ManTestX+ManTest2+2014'), 'edx/demox/DemoCourse']
+            )
+
+            if status == 'inactive':
+                self.assertDictEqual({}, parse_data)
+            else:
+                self.assertDictEqual(
+                    {
+                        'course-v1:ManTestX+ManTest2+2014': {
+                            'run_modes_count': run_modes_count,
+                            'marketing_slug': slug,
+                            'display_name': 'Manual Test',
+                            'category': 'xseries'
+                        }
+                    },
+                    parse_data
+                )
+
+    @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+    def test_program_courses_on_dashboard_without_configuration(self):
+        """If programs configuration is disabled than xseries upsell messages
+        will not appear on student dashboard.
+        """
+        CourseEnrollment.enroll(self.user, self.course.id)
+        self.client.login(username="jack", password="test")
+        response = self.client.get(reverse('dashboard'))
+        self.assertEquals(response.status_code, 200)
+        self.assertIn('Pursue a Certificate of Achievement to highlight', response.content)
+        self._assert_responses(response, 0)
+
+    @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
```

Comment:
same as above. Instead of doing this on individual tests use it on the test class